argo_rollouts/tests/test_e2e.py

```diff
@@ -0,0 +1,21 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import pytest
+
+from datadog_checks.base.constants import ServiceCheck
+from datadog_checks.dev.utils import assert_service_checks, get_metadata_metrics
+
+from .common import E2E_METRICS
+
+
+def test_e2e_openmetrics_v2(dd_agent_check):
```

Comment:
So this test checks the following:

1. interaction with the agent (can the agent invoke the integration)
2. interaction with argo rollouts (can we successfully connect to it)
3. can we submit 2 service checks
4. can we submit some metrics:
  a. some special metrics that we somehow can't get as fixtures
  b. histograms in their full form, we need agent aggregation for it

I'm worried that only points 1 and 2 are **really necessary**, but we're paying a pretty heavy price for them. Any kind of e2e test with this many assertions is likely to flake at some point.

Could we move the metrics-related assertions into unit tests? 4a. **should be** possible to turn into a fixture and for 4b imo it's sufficient to assert that we've submitted a metric with type `histogram`. Testing the agent's aggregation extensively isn't really worth introducing flaky tests imo.

I'm bringing this up only now because I've been thinking a lot about tests recently (and talking to Jose, ngl 😂 )

Reply:
@iliakur hmm okay. I get what you're saying. I also feel the e2e test is redundant since the unit test covers this already. But I think it's required to have a e2e for the integration. That said, if I were to remove 3 and 4 in your list, how would we know the agent interacted with argo or that the agent is able to load the check at all? Shouldn't I atleast assert one of the products of running the check as a means to validate that a check has ran? Or am I missing the point?

Reply:
Good question! Having an e2e test is defo important and necessary, the question is how many asserts we've got in there.

I'd say let's keep the bare minimum of assertions that we need to make sure the check ran.

Good candidates are: 

- service check
- metric that's used to automatically detect the integration

Everything else is overkill for the e2e test in this case.

Reply:
Sounds good. I'll drop the metric then and keep the service check.